server/models/views/river-and-sea-levels.js

```diff
@@ -1,287 +1,310 @@
+    station.state = getStationState(station)
+    station.group_type = getStationGroup(station)
   }
+}
 
-  formatName (name) {
-    return name.toLowerCase().replace(/(^\w{1})|(\s+\w{1})/g, letter => letter.toUpperCase())
+function ReferencedStationViewModel (referencePoint, stations) {
+  const center = turf.point([referencePoint.lon, referencePoint.lat]).geometry
+  const bbox = createBbox(stations)
+  stations.forEach(station => {
+    setStationProperties(station, center.coordinates)
+  })
+  stations.sort((a, b) => a.distance - b.distance)
+
+  const { filters, activeFilter: queryGroup } = setFilters(stations)
+
+  deleteUndefinedProperties(stations)
+
+  return {
+    stations,
+    filters,
+    queryGroup,
+    exports: {
+      placeBox: bbox,
+      bingMaps: bingKeyMaps
+    },
+    floodRiskUrl,
+    distStatement: referencePoint.distStatement,
+    pageTitle: 'Find river, sea, groundwater and rainfall levels',
+    metaDescription: 'Find river, sea, groundwater and rainfall levels in England. Check the last updated height and state recorded by the gauges.'
   }
 
-  formatExpiredTime (date) {
-    const formattedTime = moment(date).tz('Europe/London').format('h:mma')
-    const formattedDate = moment(date).tz('Europe/London').format('D MMMM')
-    return `Updated ${formattedTime}, ${formattedDate}`
+  function setStationProperties (station, referenceCoordinates) {
+    station.external_name = formatName(station.external_name)
+    station.displayData = getDisplayData(station)
+    station.latestDatetime = station.status === 'Active' ? getFormattedTime(station) : null
+    station.formattedValue = station.status === 'Active' ? formatValue(station, station.value) : null
+    station.state = getStationState(station)
+    station.group_type = getStationGroup(station)
+    station.distance = calcDistance(station, referenceCoordinates)
   }
+}
 
-  formatValue (station, val) {
-    const dp = station.station_type === 'R' ? 1 : 2
-    return parseFloat(Math.round(val * Math.pow(10, dp)) / (Math.pow(10, dp))).toFixed(dp) + (station.station_type === 'R' ? 'mm' : 'm')
+function ViewModel ({ location, place, stations, referer, queryGroup, rivers, rloiid, rainfallid, originalStation, targetArea, riverid }) {
+  let bbox, filters, activeFilter, distStatement, pageTitle, metaDescription, center, stationsBbox
+  const isEngland = place ? place.isEngland.is_england : true
+
+  if (stations && isEngland) {
+    ({ originalStation, bbox } = mapProperties(rloiid, originalStation, stations, bbox, rainfallid, targetArea, riverid))
+    stations.forEach(station => {
+      stationProperties(station, place, stations, originalStation)
+    })
+    stations.sort((a, b) => a.distance - b.distance)
+
+    // ref: https://flaviocopes.com/javascript-destructure-object-to-existing-variable/
+    ;({ filters, activeFilter } = setFilters(stations, queryGroup))
+
+    deleteUndefinedProperties(stations)
+
+    stations = isEngland ? stations : []
```

Comment:
ternary operator is inside a check that has already checked for stations

Reply:
Removed in #348